lib/dpl/provider/pypi.rb

```diff
@@ -96,7 +96,10 @@ def push_app
         context.shell "python setup.py #{pypi_distributions}"
         context.shell "twine upload -r pypi dist/*"
         context.shell "rm -rf dist/*"
-        context.shell "python setup.py upload_docs #{pypi_docs_dir_option} -r #{pypi_server}"
+        unless options[:skip_upload_docs]
+          log "Uploading documentation (skip with \"skip_upload_docs: true\")"
+          context.shell "python setup.py upload_docs #{pypi_docs_dir_option} -r #{pypi_server}"
```

Comment:
Yes - you can run it multiple times. That was the behavior before this PR anyway, and I believe I've proved that it works as expected with my tests linked in the OP.